lib/pundit/policy_finder.rb

```diff
@@ -1,4 +1,46 @@
 module Pundit
+  class ScopeResolver
+    attr_reader :policy
+    attr_reader :user
+    attr_reader :resource_scope
+
+    def initialize(user, resource_scope)
+      @user = user
+      @resource_scope = resource_scope.respond_to?(:each) ? resource_scope.last : resource_scope
```

Comment:
It depends: if you want to accept specifically both Arrays and single values then this approach is possible (Inspired by Avdi Grimm's "Confident Ruby"):

```ruby
Array(resource_scope).last
```

If this needs to support arbitrary `Enumerable`s (which is what your code currently does) then I'm not sure. I don't really understand the surrounding context of this code. 

The fact that we're having to accept data which might be of different shapes feels like a code smell, but I'm not sure we can get away from that, since the rest of the code is full of `respond_to?`.

Reply:
@laertispappas what did you decide about this?